### PR TITLE
Allow to use account with contracts on client

### LINKED ISSUE #1349: Implement and test smart contract verification
Implement neo-project/neo#1800 changes, add a test based on https://github.com/neo-project/neo-node/pull/628#issuecomment-664640875.

### PR DESCRIPTION
**Pull request by fyrchik**
Fix #1349 .

### CODE REVIEW

File: pkg/rpc/client/rpc.go [hunk 1]
```diff
@@ -499,16 +500,19 @@ func (c *Client) AddNetworkFee(tx *transaction.Transaction, extraFee int64, accs
 	}
 	size := io.GetVarSize(tx)
 	for i, cosigner := range tx.Signers {
-		if accs[i].Contract.Script == nil {
-			contract, err := c.GetContractState(cosigner.Account)
-			if err == nil {
-				if contract == nil {
-					continue
+		if accs[i].Contract.Deployed {
```
Comment (roman-khimov) [inline pkg/rpc/client/rpc.go:503]: So how do I create a wallet for deployed contract that I could use with standard CLI commands?

File: pkg/rpc/client/rpc.go [hunk 2]
```diff
@@ -499,16 +500,19 @@ func (c *Client) AddNetworkFee(tx *transaction.Transaction, extraFee int64, accs
...
+			if err == nil && res.State == "HALT" && len(res.Stack) == 1 {
+				r, err := topIntFromStack(res.Stack)
+				if err != nil || r == 0 {
+					return core.ErrVerificationFailed
 				}
-				netFee, sizeDelta := core.CalculateNetworkFee(contract.Script)
-				tx.NetworkFee += netFee
-				size += sizeDelta
+			} else {
+				return core.ErrVerificationFailed
```
Comment (roman-khimov) [inline pkg/rpc/client/rpc.go:511]: Coverage?

File: pkg/rpc/client/rpc.go [hunk 3]
```diff
@@ -499,16 +500,19 @@ func (c *Client) AddNetworkFee(tx *transaction.Transaction, extraFee int64, accs
...
 			}
+			tx.NetworkFee += res.GasConsumed
+			size += io.GetVarSize([]byte{}) * 2 // both scripts are empty
```
Comment (roman-khimov) [inline pkg/rpc/client/rpc.go:514]: In practice invocation script would probably have some signature inside.

Comment (fyrchik) [reply pkg/rpc/client/rpc.go:514]: Didn't want to overload this PR with facilities not presented in original issue.
It is easy to extend though, if we restrict ourselves to fixed-size signature.

File: pkg/rpc/client/rpc.go (hunk 2 above)
Comment (fyrchik) [reply pkg/rpc/client/rpc.go:511]: Fixed.

File: pkg/rpc/client/rpc.go (hunk 1 above)
Comment (fyrchik) [reply pkg/rpc/client/rpc.go:503]: Fixed.

File: pkg/rpc/server/testdata/verification_contract.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package testdata
+
+import (
+	"github.com/nspcc-dev/neo-go/pkg/interop/convert"
+	"github.com/nspcc-dev/neo-go/pkg/interop/runtime"
+	"github.com/nspcc-dev/neo-go/pkg/interop/util"
+)
+
+// Verify is a verification contract method.
+// It returns true iff it is signed by NbTiM6h8r99kpRtb428XcsUk1TzKed2gTc (id-0 private key from testchain).
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/rpc/server/testdata/verification_contract.go:10]: Fixed.

File: pkg/core/interop/runtime/witness.go [hunk 5]
```diff
@@ -49,6 +53,9 @@ func checkScope(d dao.DAO, tx *transaction.Transaction, v *vm.VM, hash util.Uint
 				if callingScriptHash.Equals(util.Uint160{}) {
 					return false, nil
 				}
+				if !v.Context().GetCallFlags().Has(smartcontract.AllowStates) {
+					return false, errors.New("missing AllowStates call flag")
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/core/interop/runtime/witness.go:57]: There are no existing tests, filled an issue #1366 .

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**